gui/pages.py

```diff
@@ -33,7 +36,7 @@ def __init__(self, parent_container, controller):
         self.conf_lvl = ConfidenceLevel(self)
         self.conf_lvl.render_confidence_level_widgets(3)
 
-        self.thr_lvl = ThresholdLevel(self)
+        self.thr_lvl = ThresholdLevel(self, self.bot)
```

Comment:
pass trafficbot instance to threshold to trigger a function later and use the setter